portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -25,16 +35,57 @@
 /** Servlet that returns  comments from the webpage.*/
 @WebServlet("/comments")
 public class DataServlet extends HttpServlet {
-
+    
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ArrayList<String> comments = new ArrayList<String>();
-    comments.add("I have always wanted to visit Jamaica!");
-    comments.add("I love that your college provides support for international students!");
-    comments.add("Go Bulldogs!");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    int maxCommentsToDisplay =  maxCommentsToDisplay(request);
+
+    Query query = new Query("Comment");
+    query.addSort("date", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    
+    List<Comment> commentList = new ArrayList<>();   
+    for (Entity commentEntity : results.asIterable(
+      FetchOptions.Builder.withLimit(maxCommentsToDisplay))) {
+      commentList.add(Comment.fromEntity(commentEntity));
+    }
+
+    Gson gson = new Gson();
+    String json = gson.toJson(commentList);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    String username = request.getParameter("username");
+    String comment = request.getParameter("comment");
+
+    datastore.put(Comment.toEntity(new Comment(username, new Date(), comment)));
+
     Gson gson = new Gson();
-    String json = gson.toJson(comments);
+    String json = gson.toJson(comment);
     response.setContentType("application/json;");
     response.getWriter().println(json);
+    response.sendRedirect("/");
+  }
+
+   /** Returns the max entered by the user, or default to 5. */
+  private int maxCommentsToDisplay(HttpServletRequest request) {
+    // Get the input from the form.
```

Comment:
The comment doesnt add anything to the code. its pretty clear what you're doing and the code is not weird or awkward or anything that it warrants a simple comment. Remove the comment